### PR TITLE
Fix a long-standing spec compliance bug where Synapse would accept a trailing slash on the end of `/get_missing_events` federation requests.

### PR DESCRIPTION
**Pull request by reivilibre**
I checked the logs on matrix.org and no recent requests hit this pattern.
It seems like this question has been around a long time without a good reason, so I propose we remove it... :)

### CODE REVIEW

File: synapse/federation/transport/server/federation.py [hunk 1]
```diff
@@ -549,8 +549,7 @@ async def on_POST(
 
 
 class FederationGetMissingEventsServlet(BaseFederationServerServlet):
-    # TODO(paul): Why does this path alone end with "/?" optional?
-    PATH = "/get_missing_events/(?P<room_id>[^/]*)/?"
+    PATH = "/get_missing_events/(?P<room_id>[^/]*)"
```
Comment (DMRobertson) [inline synapse/federation/transport/server/federation.py:552]: Should that `*` be a `+`? Presumably room ids must be nonempty?

Comment (reivilibre) [reply synapse/federation/transport/server/federation.py:552]: Looking at a few other examples, for identifiers we just do `*` and rely on validating them after the fact, so using `+` would be 'inconsistent'. Not sure it makes much difference either way; maybe we should also validate that they start with `!` but I don't know.

Comment (clokep) [reply synapse/federation/transport/server/federation.py:552]: I think if it was a `+` then the servlet matching code wouldn't even match to this request and you would get a generic `M_UNRECOGNIZED_REQUEST` error. Hopefully by being more graceful in matching we can give a better error about a missing parameters or something?

Comment (DMRobertson) [reply synapse/federation/transport/server/federation.py:552]: Sounds like we should leave this alone!

**Review by DMRobertson — APPROVED** [review]

**Pull request merged**